File: Programmers/Lv.1/대충 만든 자판/160586.js
function solution(keymap, targets) {
    const answer = new Array(targets.length).fill(0);
    
    for (let i = 0; i < targets.length; i++) {
        for (const target of [...targets[i]]) {
            let count = 999;
            
            for (const keys of keymap) {
                const index = keys.indexOf(target);
                if (index > -1 && index < count) {
                    count = index;
                }
            }
            
            if (count === 999) {
                answer[i] = -1;
                break;
            } else {
                answer[i] += count + 1;
            }
        }
    }
    
    return answer;
}

File: Programmers/Lv.1/대충 만든 자판/160586.java
class Solution {
    public int[] solution(String[] keymap, String[] targets) {
        int[] answer = new int[targets.length];
        
        for (int i = 0; i < targets.length; i++) {
            for (String target : targets[i].split("")) {
                int count = 999;
                
                for (String keys : keymap) {
                    int index = keys.indexOf(target);
                    if (index > -1 && index < count) {
                        count = index;
                    }
                }
                
                if (count == 999) {
                    answer[i] = -1;
                    break;
                } else {
                    answer[i] += count + 1;
                }
            }
        }
        
        return answer;
    }
}
